Microsoft.HealthVault.Fhir/Transformers/PractitionerToHealthVault.cs

```diff
@@ -0,0 +1,136 @@
+using Hl7.Fhir.Support;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static partial class PractitionerToHealthVault
+    {
+        public static ItemTypes.PersonItem ToHealthVault(this Hl7.Fhir.Model.Practitioner fhirPractitioner)
+        {
+            return fhirPractitioner.ToPerson();
+        }
+
+        private static ItemTypes.PersonItem ToPerson(this Hl7.Fhir.Model.Practitioner fhirPractitioner)
+        {
+            if (fhirPractitioner == null || fhirPractitioner.Name.IsNullOrEmpty())
+                return null;
+            
+            var practitionerName = fhirPractitioner.Name.First(); // Let's consider just the first item
+
+            //name
+            var person = new ItemTypes.PersonItem()
+            {
+                PersonType = new CodableValue("Provider", new CodedValue("1", "person-types")),
+                Name = new ItemTypes.Name()
+                {
+                    Last = practitionerName.Family,
+                    Suffix = practitionerName.Suffix.Any() ? new ItemTypes.CodableValue(practitionerName.Suffix.First()) : null,
+                    Title = practitionerName.Prefix.Any() ? new ItemTypes.CodableValue(practitionerName.Prefix.First()) : null,
+                    First = practitionerName.Given.FirstOrDefault() ?? string.Empty,
+                    Middle = practitionerName.Given.ElementAtOrDefault(1) ?? string.Empty,
+                },
+                ContactInformation = new ContactInfo()
+            };            
+            if (!string.IsNullOrEmpty(practitionerName.Text))
+            {
+                person.Name.Full = practitionerName.Text;
+            }
+                              
+
+            //address
+            if (!fhirPractitioner.Address.IsNullOrEmpty())
+            {
+                foreach (var address in fhirPractitioner.Address)
+                {
+                    var hvAddress = new ItemTypes.Address();
+
+                    foreach (var line in address.Line)
+                    {
+                        hvAddress.Street.Add(line);
+                    }
+                    hvAddress.City = address.City;
+                    hvAddress.State = address.State;
+                    hvAddress.County = address.District;
+                    hvAddress.PostalCode = address.PostalCode;
+                    if (!string.IsNullOrEmpty(address.Country))
+                    {
+                        hvAddress.Country = address.Country;
+                    }
+                        
+
+                    hvAddress.Description = address.Text;
+                    hvAddress.IsPrimary = address.GetBoolExtension(HealthVaultExtensions.IsPrimary);
+
+                    person.ContactInformation.Address.Add(hvAddress);
+                }
+            }
+
+            //telecom
+            if (!fhirPractitioner.Telecom.IsNullOrEmpty())
```

Comment:
Very similar code is in 'Transformers/PatientToContact.cs' - is there a way to abstract the code and reuse?

Reply:
Fixed: Added an ToHealthVault extension for Fhir ContactPoint. Introduced abstraction only for the conversion of a ContactPoint.